background.ts

```diff
@@ -747,19 +771,248 @@ function newK8sManager() {
   return mgr;
 }
 
+type validationFunc = (name: string, value: string|boolean) => string;
```

Comment:
I see TypeScript doesn't get too hung up on function types -- I defined that on version 1, never changed it, and it's out of date. Will fix.

Changing both `validate` and `verify` to `check...` because it's shorter. Although I still like the name `validationFunc`.

As for making `errors` a writable parameter, I like the way it works. I picked up this technique way back when I learned Lisp and Scheme. The errors are sort of a side-effect, whereas the `changesNeeded` boolean works out fine as a return value with the near-functional programming that's going on here. Really, if you look at the code and try to have all the checkers recursively return and build something like `{ changeNeeded:boolean, errors:string[]}` the code is more cumbersome.

Now that all the validation/checking is in a separate module the code in `background.ts` doesn't have to worry about how `errors` is calculated.